tools/dotcms-cli/cli/src/main/java/com/dotcms/cli/common/DotExceptionHandler.java

```diff
@@ -0,0 +1,50 @@
+package com.dotcms.cli.common;
+
+import com.dotcms.cli.command.DotCommand;
+import io.quarkus.arc.Arc;
+import java.util.Optional;
+import picocli.CommandLine;
+import picocli.CommandLine.IExecutionExceptionHandler;
+import picocli.CommandLine.ParseResult;
+
+/**
+ * This class is meant to handle exceptions that occur during command execution
+ * It will delegate the handling of the exception to the OutputOptionMixin
+ * If the OutputOptionMixin is not available, it will print the error message to the console
+ */
+public class DotExceptionHandler implements IExecutionExceptionHandler {
+
+    @Override
+    public int handleExecutionException(Exception ex, CommandLine commandLine, ParseResult parseResult) throws Exception {
+        String commandName = "UNKNOWN" ;
+        boolean isShowErrors = false;
+        final Object object = commandLine.getCommand();
+        //This takes the parseResult and unwraps the subcommands to get the command that was executed
+        final Optional<CommandsChain> chain = SubcommandProcessor.process(parseResult);
+        if (chain.isPresent()) {
```

Comment:
Maybe you can use ifPresent method instead of use if sentence.

Reply:
The thing about using it as ifPresent is that whatever gets modified from within that block needs to be final 
So doing something like this:
chain.ifPresent(commandsChain -> {
            commandName = commandsChain.command();
            isShowErrors = commandsChain.isShowErrorsAny();
        });
presents that problem